bundles/org.openhab.binding.fmiweather/src/main/java/org/openhab/binding/fmiweather/internal/ForecastWeatherHandler.java

```diff
@@ -172,6 +153,71 @@ protected void updateChannels() {
         }
     }
 
+    private void updateHourlyChannels(FMIResponse response, Location location) throws FMIUnexpectedResponseException {
+        for (Channel channel : getThing().getChannels()) {
+            ChannelUID channelUID = channel.getUID();
+            if (CHANNEL_GROUP_FORECAST.equals(channelUID.getGroupId())) {
+                // Skip time series group
+                continue;
+            }
+            int hours = getHours(channelUID);
+            int timeIndex = getTimeIndex(hours);
+            if (channelUID.getIdWithoutGroup().equals(CHANNEL_TIME)) {
+                // All parameters and locations should share the same timestamps. We use temperature to figure out
+                // timestamp for the group of channels
+                final String field = ForecastRequest.PARAM_TEMPERATURE;
+                Data data = unwrap(response.getData(location, field),
+                        "Field %s not present for location %s in response. Bug?", field, location);
+                updateEpochSecondStateIfLinked(channelUID, data.timestampsEpochSecs[timeIndex]);
+            } else {
+                String field = getDataField(channelUID);
+                Unit<?> unit = getUnit(channelUID);
+                if (field == null) {
+                    logger.error("Channel {} not handled. Bug?", channelUID.getId());
+                    continue;
+                }
+                Data data = unwrap(response.getData(location, field),
+                        "Field %s not present for location %s in response. Bug?", field, location);
+                updateStateIfLinked(channelUID, data.values[timeIndex], unit);
+            }
+        }
+    }
+
+    private void updateTimeSeriesChannels(FMIResponse response, Location location)
+            throws FMIUnexpectedResponseException {
+        for (Channel channel : getThing().getChannelsOfGroup(CHANNEL_GROUP_FORECAST)) {
+            ChannelUID channelUID = channel.getUID();
+            if (CHANNEL_TIME.equals(channelUID.getIdWithoutGroup())) {
+                // All parameters and locations should share the same timestamps. We use temperature to figure out
+                // timestamp for the group of channels
+                final String field = ForecastRequest.PARAM_TEMPERATURE;
+                Data data = unwrap(response.getData(location, field),
+                        "Field %s not present for location %s in response. Bug?", field, location);
+                updateEpochSecondStateIfLinked(channelUID, data.timestampsEpochSecs[0]);
+                continue;
+            }
+            String field = getDataField(channelUID);
+            Unit<?> unit = getUnit(channelUID);
+            if (field == null) {
+                logger.error("Channel {} not handled. Bug?", channelUID.getId());
```

Comment:
```suggestion
                logger.warn("Channel {} not handled. Bug?", channelUID.getId());
```

Reply:
Isn't error correct?

> in case some code fails irrecoverably and the user should report it as a severe bug.

https://www.openhab.org/docs/developer/guidelines.html#f-logging

Reply:
When this happens it means that `CHANNEL_TO_FORECAST_FIELD_NAME_AND_UNIT` is out of sync with defined channels. This is a bug, and the binding will not work correctly. This is equivalent to a binding throwing `IllegalStateException`, which would have a more severe outcome. So I agree `error` is correct. It is also consistent with the already existing error handling.

Reply:
Was looking at warn
>…to inform the user that something seems to be wrong in the overall setup, but the system can nonetheless function as normal

The system has no problem and even the binding remains to work, scope is limited to a channel.  As it is also a bug, error would give it more attention, let’s stick to that